### DISCUSSION TITLE
Connecting to Database, Photoprism installed using Docker Compose

### DISCUSSION THREAD

**Discussion by jshokrollahi** [photoprism/photoprism#4628] [Q&A] [answered]
Hello all,

I have installed Photoprism using Docker Compose on a Raspberry. 
I have also installed mariadb client. But I do not know how to connect 
to the database. 

I found this link, but first, I do not know if my issue is the same?
https://www.reddit.com/r/photoprism/comments/13d3r5o/how_do_access_photoprism_database/
If yes, I do not know where to give the port map mentioned on the link.

Thanks and best regards

Jamshid

**Comment by omerdduran** [comment, marked as answer]
Hello. To connect to MariaDB running in Docker, you need to do one of the following. You can choose whichever one suits your needs:

- Open the database port (3306) to your host system or
- Connecting directly through the Docker container

Option 1 - Expose the port in docker-compose.yml:
```yaml
services:
  mariadb:
    # ... other config ...
    ports:
      - "3306:3306"
```

Option 2 - Connect through the Docker container:
```bash
# Connect to MariaDB through Docker
docker compose exec mariadb mariadb -u username -p database-name
# Usually like this 
docker compose exec mariadb mariadb -u photoprism -p photoprism
```

Important security notes:
1. Only expose port 3306 if you really need external database access
2. Make sure to use strong passwords if exposing the database port

**Reply by jshokrollahi** [reply]
Hello,

Thank you very much for the answer! 
I have tried both of them and they work!

Best regards,
Jamshid
